.github/workflows/workspace-tests.yaml

```diff
@@ -155,24 +169,62 @@ jobs:
                 continue
               fi
 
+              # Extract config content for environment variable validation
+              CONFIG_CONTENT=$(yq -o=yaml '.spec.appConfigExamples[0].content' "$METADATA_FILE")
+              
+              # Extract dynamicPlugins section for environment variable checking
+              DYNAMIC_PLUGINS_CONFIG=$(yq -o=yaml '.spec.appConfigExamples[0].content.dynamicPlugins' "$METADATA_FILE" 2>/dev/null || echo "")
+              
+              # Extract environment variables from dynamicPlugins config (pattern: ${VAR} or $VAR)
+              ENV_VARS=""
+              if [ -n "$DYNAMIC_PLUGINS_CONFIG" ] && [ "$DYNAMIC_PLUGINS_CONFIG" != "null" ]; then
+                ENV_VARS=$(echo "$DYNAMIC_PLUGINS_CONFIG" | grep -oE '\$\{[A-Z_][A-Z0-9_]*\}|\$[A-Z_][A-Z0-9_]*' | sed 's/\${//; s/}//; s/^\$//' | sort -u)
+              fi
+              
+              if [ -n "$ENV_VARS" ]; then
+                # Config contains environment variables
+                if [ -z "$WORKSPACE_ENV_CONTENT" ]; then
+                  echo "  ::warning::Config for $PLUGIN_NAME contains environment variables but workspace test.env is missing. Tests will be skipped."
+                  SKIP_TESTS_MISSING_ENV="true"
+                  PLUGINS_SKIPPED_MISSING_ENV=$((PLUGINS_SKIPPED_MISSING_ENV + 1))
+                  continue
+                else
+                  # Validate all ENVs are present in merged env file
+                  MISSING_ENVS=()
+                  while IFS= read -r env_var; do
+                    [ -z "$env_var" ] && continue
+                    if ! echo "$WORKSPACE_ENV_CONTENT" | grep -qE "^[[:space:]]*${env_var}[[:space:]]*="; then
+                      MISSING_ENVS+=("$env_var")
+                    fi
+                  done <<< "$ENV_VARS"
+                  
+                  if [ ${#MISSING_ENVS[@]} -gt 0 ]; then
+                    echo "  ::error::Environment variables missing from test.env: ${MISSING_ENVS[*]}"
+                    echo "  Config for $PLUGIN_NAME references these environment variables but they are not defined in $WORKSPACE_ENV_FILE."
+                    exit 1
+                  fi
+                fi
+              fi
+              # If no ENVs in config, continue regardless of env file existence
+
               STRIPPED=$(echo "$PACKAGE_NAME" | sed 's|^@||; s|/|-|')
               echo "- package: \"oci://ghcr.io/${IMAGE_PATH_AND_PLUGIN}:${NEW_TAG}!${STRIPPED}\"" >> "$OUT_FILE"
               echo "  disabled: false" >> "$OUT_FILE"
               echo "  pluginConfig:" >> "$OUT_FILE"
-              # Append the configs.default block indented by 4 spaces under pluginConfig
-              yq -o=yaml '.configs.default' "$METADATA_FILE" | sed 's/^/    /' >> "$OUT_FILE"
+              # Append the first appConfigExamples content block indented by 4 spaces under pluginConfig
+              echo "$CONFIG_CONTENT" | sed 's/^/    /' >> "$OUT_FILE"
```

Comment:
Seems to me that using jq & yq, which afaik are included by default in the default ubuntu runner, would be more robust ?